### DISCUSSION TITLE
Show custom icon for folder/containers

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#14838] [Features and ideas]
Currently both root nodes/folders, folder/containers and content types like Folder media type use the same `icon-folder` icon.
I think its an UX/UI issue because the nodes looks the same, but doesn't act the same or have same functionality.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/5cc5ea3d-c20a-446e-ba39-6be53227249d)

As a minimum I think we should register a folder icon used for folders/containers, that isn't available in the icon picker.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/ddf12d6b-0afd-4756-87ea-da4b0a6f8887)

**Comment by nul800sebastiaan** [comment]
As a workaround for now you can use emoji for each folder name.

**Reply by bjarnef** [reply]
Sure, not sure how well it works in different browsers though 😜
and potientially emoji can also be used for Folder media type.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/da82403e-af52-409e-a74d-17410564e1f5)

**Reply by bjarnef** [reply]
An alternative version.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/f5d5f1da-fe7f-4b75-9043-372ff1b6d653)

**Reply by nul800sebastiaan** [reply]
I'd put them at the beginning of the name so you can scan more easily. 

It's not a solution, but I am not sure folder customizations is going to be a priority any time soon, so this is just a little way to help. Of course you can also hook into the tree rendering notification and based on some rules replace the icon. Might be a good package even!

**Reply by bjarnef** [reply]
Yes, it may be something to try out.
Meanwhile I have added support for member type container 😁 
https://github.com/umbraco/Umbraco-CMS/pull/14833

**Comment by markadrake** [comment]
I wrote something to apply custom icons to content nodes in the **content** section.
Do the same extensibility points exist for the **settings** section?

![image](https://github.com/umbraco/Umbraco-CMS/assets/713355/b6d15446-a36b-4e6e-8711-76e705b17570)

![image](https://github.com/umbraco/Umbraco-CMS/assets/713355/4ff8e3a9-29c7-439e-8a3e-ef1bcf55e284)

![image](https://github.com/umbraco/Umbraco-CMS/assets/713355/b5f844f7-074f-49b5-8cbf-5ae23f12ff73)

Let me know if there's any interest in exploring it.

**Comment by markadrake** [comment]
Figuring out how to reload the correct section of the tree has been the worst part of this update. Otherwise, I think I was able to quickly adapt my custom icons (for content nodes) to all tree nodes. 

<img width="327" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/713355/975cb198-edd5-48a7-8841-9b08bff5be83">

<img width="283" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/713355/cf4ad22d-b42a-4d86-a871-8f2595f88b2e">

<img width="246" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/713355/8e35b6a5-93f0-4a7b-88b1-500791180507">

<img width="255" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/713355/0e5d2e5f-7417-4419-bb28-a03da4a4d290">

<img width="323" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/713355/78163c09-1bb7-46e7-9613-9c4ecb85d086">
